### DISCUSSION TITLE
How to retrieve a list of a user's games?

### DISCUSSION THREAD

**Discussion by qs-arno** [ValvePython/steam#381] [Q&A] [answered]
I've got two ideas for tools I'd like to write/improve, which require similar information; I'd like to be able to:

1. ...retrieve a list of my own owned products (games/DLC/videos/etc.), with purchase dates if possible, and
2. ...retrieve a list of another user's games (don't care about purchase dates...DLC info would be nice).

I'm aware that you can pull down a user's game list without any authentication in XML format straight from the Web site, but I'm trying to improve on that for two reasons:

1. In the case of my own game list, this method will break if I ever decide to make my game list non-public.
2. In the case of my friends' games, that method doesn't work for my friends whose game lists have friends-only visibility.

This seems like something that should be straightforward, but for the life of me I can't find anything in the docs that looks right.  I've already got the API up and running (I'm using it to redeem keys), so I don't think I need help with general configuration...just this particular call.  Does it exist?

**Comment by rossengeorgiev** [comment, marked as answer]
After login in with `SteamClient` you receive a license list with all the apps that the account owns. You can then inspect get all the owned games. Here is `steamctl apps licenses list` implementation https://github.com/ValvePython/steamctl/blob/4d6b19a921a4db521ed6e0d05a1dae2458441a9a/steamctl/commands/apps/gcmds.py#L179-L227

I suppose you could also scrape https://store.steampowered.com/account/history/

Steam community xml or via Steam web api, but depends on their privacy settings. Either public, or friend only if you are friends with the user. https://steamapi.xpaw.me/#IPlayerService/GetOwnedGames

**Reply by qs-arno** [reply]
Thanks for the info!  I haven't really explored the Web API yet, but this looks like a great way to get started.
